### PR TITLE
Fea/support cmake cuda architectures rapids value

### LINKED ISSUE #314: Expand CMAKE_CUDA_ARCHITECTURE to support `RAPIDS` ( equal to `ALL` )(
**Is your feature request related to a problem? Please describe.**
CMake 3.23+ now offers support for `all` and `all-major` as special keywords to `CMAKE_CUDA_ARCHITECTURES`. This means that rapids-cmake overload on `ALL` now creates confusion and breaks expectations ( https://github.com/trxcllnt/rapids-compose/issues/84 https://gitlab.kitware.com/cmake/cmake/-/issues/23739, https://github.com/rapidsai/rapids-cmake/issues/104 )

**Describe the solution you'd like**
What we should do is offer a new magic value named `RAPIDS` which will be the defacto way to build for the subset of CUDA architectures that RAPIDS projects need to support ( a subset of all-major ).

### PR DESCRIPTION
**Pull request by robertmaynard**
## Description
CMake 3.23+ now offers support for `all` and `all-major` as special keywords to `CMAKE_CUDA_ARCHITECTURES`. This means that rapids-cmake overload on `ALL` now creates confusion and breaks expectations ( https://github.com/trxcllnt/rapids-compose/issues/84 https://gitlab.kitware.com/cmake/cmake/-/issues/23739, https://github.com/rapidsai/rapids-cmake/issues/104 ) 

This PR introduces the new magic keyword of `RAPIDS` which behaves the same as `ALL`, and will become the new apporved approach for projects that want to build for RAPIDS supported platforms.

Fixes #314

More details on the plans of deprecation for `ALL` can be found at: https://github.com/rapidsai/rapids-cmake/issues/318

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/rapids-cmake/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The `cmake-format.json` is up to date with these changes.

### CODE REVIEW

File: rapids-cmake/cuda/init_architectures.cmake [hunk 1]
```diff
@@ -79,21 +71,22 @@ function(rapids_cuda_init_architectures project_name)
...
+    set(cuda_arch_mode "RAPIDS")
   elseif(CMAKE_CUDA_ARCHITECTURES STREQUAL "" OR CMAKE_CUDA_ARCHITECTURES STREQUAL "NATIVE")
     set(cuda_arch_mode "NATIVE")
   elseif(NOT (DEFINED ENV{CUDAARCHS} OR DEFINED CMAKE_CUDA_ARCHITECTURES))
-    set(cuda_arch_mode "ALL")
+    set(cuda_arch_mode "RAPIDS")
   endif()
 
   # This needs to be run before enabling the CUDA language since RAPIDS supports the magic string of
   # "ALL"
```
Comment (bdice) [inline rapids-cmake/cuda/init_architectures.cmake:88]: This comment seems inaccurate now -- but I'm not entirely sure about the purpose of this PR. The PR description is empty.

**Review by bdice — CHANGES_REQUESTED** [review]
Why is this change needed? There is no description or linked issue that I can see.

File: rapids-cmake/cuda/init_architectures.cmake (hunk 1 above)
Comment (robertmaynard) [reply rapids-cmake/cuda/init_architectures.cmake:88]: Sorry about that. I have updated the description of the PR to explain what the end goal is. But in short we are moving away from the `ALL` keyword as the term is loaded now that `all` is handled by CMake / nvcc and produces a different set of cuda arch values.

The comment is stale now, as it should state that RAPIDS supports the magic keywords of `ALL` and `RAPIDS` ( both map to a `cuda_arch_mode` value of `RAPIDS` ). I have pushed an update to fix the comment

**Review by bdice — APPROVED** [review]
Looks good!

**Comment by robertmaynard** [comment]
@gpucibot merge

**Pull request merged**